.github/workflows/linting.yml

```diff
@@ -0,0 +1,42 @@
+name: Linting
+
+# Always run on Pull Requests as then these checks can be marked as required.
+on:
+  push:
+    branches:
+      - master
+      - 'feature/*'
+      - 'release/*'
+  pull_request:
+
+jobs:
+  # Run isort on the tree.
+  # This checks .py files only so misses SConstruct and SConscript files are not checked, rather
+  # for these files check them afterwards.  The output-filter will not be installed for this part
+  # so regressions will be detected but not annotated.
+  isort:
+    name: Python isort
```

Comment:
This check is failing on this PR:
https://github.com/daos-stack/daos/actions/runs/6751564908/job/18355826582?pr=13270

Reply:
Can we address this in a separate PR? It would be nice to get this landed so we can merge master into the branch, even though it will fail these python checks.

Reply:
Yeah, since this is a feature branch I'm okay with that